packages/cli/src/tools/envinfo.ts

```diff
@@ -2,20 +2,39 @@
 import envinfo from 'envinfo';
 import {EnvironmentInfo} from '../commands/doctor/types';
 
-export default async function getEnvironmentInfo() {
-  return JSON.parse(
-    await envinfo.run(
-      {
-        Binaries: ['Node', 'Yarn', 'npm', 'Watchman'],
-        IDEs: ['Xcode', 'Android Studio'],
-        SDKs: ['iOS SDK', 'Android SDK'],
-        npmPackages: ['react', 'react-native', '@react-native-community/cli'],
-        npmGlobalPackages: ['*react-native*'],
-      },
-      {
-        json: true,
-        showNotFound: true,
-      },
-    ),
-  ) as EnvironmentInfo;
+/**
+ * Returns information about the running system.
+ * If `json === true`, or no options are passed,
+ * the return type will be an `EnvironmentInfo`.
+ * If set to `false`, it will be a `string`.
+ */
+async function getEnvironmentInfo(): Promise<EnvironmentInfo>;
+async function getEnvironmentInfo(json: true): Promise<EnvironmentInfo>;
+async function getEnvironmentInfo(json: false): Promise<string>;
+async function getEnvironmentInfo(
+  json = true,
+): Promise<string | EnvironmentInfo> {
+  const options = json
+    ? {json: true, showNotFound: true}
+    : {json: false, showNotFound: true};
```

Comment:
We can have a single object and use json value because it's already boolean

Reply:
Done!